cmd/snap-exec/main.go

```diff
@@ -177,31 +191,22 @@ func execApp(snapApp, revision, command string, args []string) error {
 
 	// strings.Split() is ok here because we validate all app fields and the
 	// whitelist is pretty strict (see snap/validate.go:appContentWhitelist)
-	// (see also overlord/snapstate/check_snap.go's normPath)
-	tmpArgv := strings.Split(cmdAndArgs, " ")
-	cmd := tmpArgv[0]
-	cmdArgs := expandEnvCmdArgs(tmpArgv[1:], osutil.EnvMap(env))
+	// (see also snap/container.go's normPath). No quotes are
+	// supported, so whitespace is actually a delimiter.
+	cmd := strings.Split(cmdAndArgs, " ")
```

Comment:
at this point, wouldn't it be better to modify `findCommand` to return a `[]string` directly?

Reply:
This is creating a new list of the command split by whitespace, which is quite a bit different than what `findCommand` would return if it was a list. Turning that list into this one would probably involve flattening and re-splitting, which is exactly what we do by having `findCommand` return a string. Are you simply saying you want to see the whitespace split happening in `findCommand` instead? That changes quite a few tests as well.

Reply:
findCommands used to return a string that needed to be split by spaces to make it an argument for exec. Let's call this string the OC, for original command.
findCommands now takes a list of things (the chain), which should not contain spaces, and appends to it the OC, which might contain spaces, and then joins them by spaces; on returning, we then need to split the whole thing by spaces again. I object to that, as it seems wasteful.

One approach would be to leave findCommand as it was, and take its output, split it by spaces, and append that to the chain list. This would presumably leave all the tests for findCommand alone, and  we could introduce a new helper for the chaining bit, which could be tested separately.

WDYT?

Reply:
Agreed, removing whitespace support from the chain (i.e. disallowing args) changes this. Let me poke about.

Reply:
Done and pushed.